Makefile

```diff
@@ -3,11 +3,12 @@ BUILD_FILES = $(shell go list -f '{{range .GoFiles}}{{$$.Dir}}/{{.}}\
 
 GH_VERSION ?= $(shell git describe --tags 2>/dev/null || git rev-parse --short HEAD)
 DATE_FMT = +%Y-%m-%d
-ifdef SOURCE_DATE_EPOCH
-    BUILD_DATE ?= $(shell date -u -d "@$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u "$(DATE_FMT)")
-else
-    BUILD_DATE ?= $(shell date "$(DATE_FMT)")
-endif
+BUILD_DATE ?= $(shell \
+	2>/dev/null \
+	date -u -j -f %s "${SOURCE_DATE_EPOCH}" "${DATE_FMT}" || \
```

Comment:
If I try the case when SOURCE_DATE_EPOCH isn't set (the default case), then this command returns on my macOS (BSD `date`):
```sh
date -u -j -f %s "" "+%Y-%m-%d" #=> "1970-01-01"
```

Due to the command exiting with status 0, the shell pipeline will short-circuit here, and the wrong date will get embedded. Do you think this could be the case of keeping `ifdef SOURCE_DATE_EPOCH`?

Reply:
Ah, I failed to check that. I'll come up with an alternative. I would rather not rely on ifdefs if possible.

Reply:
I've fixed the issue by adding a test before the command. It probably should be considered a bug on FreeBSD/Darwin that it behaves this way, though.

Reply:
Sure! But now you've moved the original `ifdef` into the `shell` expression via what's basically a ternary operator. I've found the previous approach easier to read. Is there a reason you're opposed to `ifdef` in Makefile?

Reply:
I would like to make it portable, one step at a time; conditionals in general can't be harmonized, though, so it would be best to remove them.